tests/integration/test_active_learning.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Integration tests for various forms of active learning implemented in Trieste.
+"""
+
+from __future__ import annotations
+
+import gpflow
+import numpy.testing as npt
+import pytest
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from tests.util.misc import random_seed
+from trieste.acquisition.function import ExpectedFeasibility, PredictiveVariance
+from trieste.acquisition.rule import AcquisitionRule, EfficientGlobalOptimization
+from trieste.bayesian_optimizer import BayesianOptimizer
+from trieste.data import Dataset
+from trieste.models import TrainableProbabilisticModel
+from trieste.models.gpflow import GaussianProcessRegression
+from trieste.objectives import BRANIN_SEARCH_SPACE, branin, scaled_branin
+from trieste.objectives.utils import mk_observer
+from trieste.space import SearchSpace
+from trieste.types import TensorType
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "num_steps, acquisition_rule",
+    [
+        (80, EfficientGlobalOptimization(PredictiveVariance())),
+    ],
+)
+def test_optimizer_learns_scaled_branin_function(
+    num_steps: int, acquisition_rule: AcquisitionRule[TensorType, SearchSpace]
+) -> None:
+    """
+    Ensure that the objective function is effecitvely learned, such that the final model
+    fits well and predictions are close to actual objective values.
+    """
+
+    search_space = BRANIN_SEARCH_SPACE
+
+    def build_model(data: Dataset) -> TrainableProbabilisticModel:
+        variance = tf.math.reduce_variance(data.observations)
+        kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=[0.2, 0.2])
+        prior_scale = tf.cast(1.0, dtype=tf.float64)
+        kernel.variance.prior = tfp.distributions.LogNormal(
+            tf.cast(-2.0, dtype=tf.float64), prior_scale
+        )
+        kernel.lengthscales.prior = tfp.distributions.LogNormal(
+            tf.math.log(kernel.lengthscales), prior_scale
+        )
+        gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+        gpflow.set_trainable(gpr.likelihood, False)
+
+        return GaussianProcessRegression(gpr)
+
+    num_initial_points = 20
```

Comment:
can we start with fewer? 20 points is already a lot, so it's possible that the `PredictiveVariance` doesn't do much.

Or we could check if the error decreases? this might be a bit tricky though...

Reply:
sure, we can try starting with fewer points

I thought about doing the decrease in error test here but concluded that would be more suitable as a unit test